lib/client.js

```diff
@@ -1,19 +1,85 @@
-var version      = require('version'),
-    Promise      = require('../vendor/ayepromise'),
-    requestCount = 0;
+var PROMISE_TIMEOUT = 5000, // 5 seconds
+    ZAF_EVENT       = /^zaf\./,
+    version         = require('version'),
+    Promise         = window.Promise || require('../vendor/ayepromise'),
+    pendingPromises = {},
+    promiseCount    = 0,
+    requestCount    = 0;
+
+function rawPostMessage(client, msg, forceReady) {
+  if (client.ready || forceReady) {
+    client._source.postMessage(msg, client._origin);
+  } else {
+    client.on('app.registered', rawPostMessage.bind(null, client, msg));
+  }
+}
+
+function removePromise(id) {
+  delete pendingPromises[id];
+}
+
+function isValidEvent(client, event) {
+  return client._origin === event.origin && client._source === event.source;
+}
+
+function messageHandler(client, event) {
+  if (!isValidEvent(client, event)) { return; }
+
+  var data = event.data;
+
+  if (!data) { return; }
```

Comment:
I think so, it was in the previous message handler too. Basically we need to filter out any postMessages that aren't in the format we expect (i.e. from other apps/sdks/etc).